module_system/installer.ts

```diff
@@ -185,9 +185,20 @@ function getModuleApiVersionFor(moduleName: string): string {
     return findDepVersionInPackageJson(moduleApiDepName, pkgJsonStr);
 }
 
+// A list of Module API versions that are supported in addition to the currently installed one
+// defined in the package.json. This is necessary because semantic versioning is applied to both
+// the Module-side surface of the API and the Client-side surface of the API. So breaking changes
+// in the Client-side surface lead to a major bump even though the Module-side surface stays
+// compatible. We aim to not break the Module-side surface so we maintain a list of compatible
+// older versions.
+const backwardsCompatibleMajorVersions = ["1.0.0"];
```

Comment:
This seems fine to me for now but I wonder if this wouldn't easily get unwieldy in future? Imagine we end up having a couple more backwards compatible versions in this list. Wouldn't the act of deciding whether or not a new version is backwards-compatible with all of the versions listed here become ever more cumbersome?

If this is a problem with differentiating between the client-facing and the module-facing API version, could we somehow define separate version numbers for these two?

CC @t3chguy / @richvdh

Reply:
I think one issue here is that `@matrix-org/react-sdk-module-api` (and also `matrix-widget-api`) are both no APIs that should be updated in an automated fashion because they require manual changes in the `matrix-react-sdk` most of the time.

Reply:
@Johennes: we discussed this in #element-dev yesterday. I suggested this approach as a pragmatic solution to the compatibility problem for now. Changing the way we manage API versions altogether would involve a bunch more thinking and work, and we can do it in future if and when this approach becomes unmanageable.

Reply:
Yes, I've followed that discussion and agreed that this seems fine for the moment. Point taken on us returning to this once it does actually become unwieldy.